### DISCUSSION TITLE
fish2rect: fisheye to rectangle/square image conversion (can we reverse it to make 2D to VR?)

### DISCUSSION THREAD

**Discussion by gituser123456789000** [nagadomi/nunif#152] [General]
I saw this a while back and looked at their preview images in VR and it looked decent from what I remember.. but I left a message there if they could do they opposite and turn regular images into fisheye. Thee hasn't been any response.If we could do the opposite, then we could do our 2D to 3D conversions on top of this and have it potentially be more immersive. Maybe that'd be decent enough 2D to VR conversion.. or at least an interesting option to have in iw3, similar to being able to create anaglyphs.

Are you able to make a reverse of this process?

https://github.com/ku6ryo/fish2rect

![fish](https://github.com/nagadomi/nunif/assets/156575428/93ae030c-2751-4371-88a7-68ea81898fa5)
![result](https://github.com/nagadomi/nunif/assets/156575428/fc16acc8-0540-4413-8f40-d47ebefc58ae)

**Comment by gituser123456789000** [comment]
here they are in anaglyph..
![fish_Any_V2_L_518_TTA_row_flow_v3_d30_c00_di0_fs0_ipd0_redcyan_half-color](https://github.com/nagadomi/nunif/assets/156575428/ec6cef3a-6c2d-4343-8471-b02315a14b87)
![result_Any_V2_L_518_TTA_row_flow_v3_d30_c00_di0_fs0_ipd0_redcyan_half-color](https://github.com/nagadomi/nunif/assets/156575428/ff034d90-755f-4e54-92e7-d8730722aae8)

**Comment by nagadomi** [comment]
iw3 already has VR180 (VR90) output option (equirectangular projection).
That doesn't improve anything as you can see if you try it.

An important aspect of VR is that the size of objects in the video is actual size.
There is no zoom-in/out and people do not change their size.

**Reply by gituser123456789000** [reply]
VR90 result cropped and anaglyphed (obviously, it's supposed to be viewed in VR, not anaglyph, but just showing)

![result vr90](https://github.com/nagadomi/nunif/assets/156575428/152295e6-f4e9-45dd-8e3a-f34afb1f6445)
![result vr90_Any_V2_L_512_TTA_row_flow_v3_d20_c10_di0_fs0_ipd0_redcyan_half-color](https://github.com/nagadomi/nunif/assets/156575428/21d0dbec-7cfc-4a1e-a551-8533d12f4505)

I think a fisheye option would be nice to see. I'm going to try messing with fish2rec.. but I probably won't even be able to get it running, let alone reverse the processing. I'll take a look at what I can do to try making it rec2fish.

**Comment by gituser123456789000** [comment]
Edit: the vr90 version I accidentally did different settings than the others.. it was 1.0 convergence with 2 strength while the others were 0.0 convergence with 3 strength.. here's a version to match the others:

![result vr90_Any_V2_L_518_TTA_row_flow_v3_d30_c00_di0_fs0_ipd0_redcyan_half-color](https://github.com/nagadomi/nunif/assets/156575428/83227b65-a587-4f24-98f1-a6e162976081)

**Comment by nagadomi** [comment]
I am not sure what you are assuming,
any image/scene can be input to iw3, but for most images that conversion will not work.
